QModManager/Patching/Patcher.cs

```diff
@@ -0,0 +1,171 @@
+                    Logger.Fatal("Please make sure that you ran Subnautica from Steam/Epic/Discord, and not from the executable file!");
+
+                    Dialog.Show("A fatal error has occurred. QModManager could not be initialized.", Dialog.Button.SeeLog, Dialog.Button.Disabled, false);
+
+                    return;
+                }
+
+                try
+                {
+                    IOUtilities.LogFolderStructureAsTree();
+                }
+                catch (Exception e)
+                {
+                    Logger.Error("There was an error while trying to display the folder structure.");
+                    Logger.Exception(e);
+                }
+
+                QModHooks.Load();
+
+                PirateCheck.CheckIfPirate(Environment.CurrentDirectory);
+
+                var gameDetector = new GameDetector();
+
+                if (!gameDetector.IsValidGameRunning)
+                    return;
+
+                CurrentlyRunningGame = gameDetector.CurrentlyRunningGame;
+
+                PatchHarmony();
+
+                if (NitroxCheck.IsInstalled)
+                {
+                    Logger.Fatal($"Nitrox was detected!");
+
+                    Dialog.Show("Both QModManager and Nitrox detected. QModManager is not compatible with Nitrox. Please uninstall one of them.", Dialog.Button.Disabled, Dialog.Button.Disabled, false);
+
+                    return;
+                }
+
+                VersionCheck.CheckForUpdates();
+
+                Logger.Info("Started loading mods");
+
+                AddAssemblyResolveEvent();
+
+                var modFactory = new QModFactory();
+                PairedList<QMod, ModStatus> modsToLoad = modFactory.BuildModLoadingList(QModBaseDir);
+
+                var initializer = new Initializer(CurrentlyRunningGame);
+                initializer.InitializeMods(modsToLoad);
+
+                QModHooks.OnLoadEnd?.Invoke();
+
+                int loadedMods = 0;
+                int erroredMods = 0;
+                foreach (Pair<QMod, ModStatus> mod in modsToLoad)
+                {
+                    if (mod.Key.IsLoaded)
+                        loadedMods++;
+                    else
+                        erroredMods++;
+                }
+
+                ErrorModCount = erroredMods;
+
+                Logger.Info($"Finished loading QModManager. Loaded {loadedMods} mods");
+
+                if (ErrorModCount > 0)
+                    Logger.Info($"A total of {ErrorModCount} mods failed to load");
+
+                SummaryLogger.LogSummaries(modsToLoad);
+            }
+            catch (FatalPatchingException pEx)
+            {
+                Logger.Fatal($"FATAL PATCHING EXCEPTION! - Patching ended");
+                Logger.Exception(pEx);
+
+                Dialog.Show("A fatal patching exception has been caught. QModManager could not be initialized.", Dialog.Button.SeeLog, Dialog.Button.Disabled, false);
```

Comment:
Upon clicking any button, the dialog will go away, which is not what we want. `Dialog` will have to be updated to allow dialogs to remain on screen even if a button was clicked.